### PR TITLE
Minor improvements to sinkhole.py

### PR DESCRIPTION
**Pull request by frinzell**
- Clean-up `import`s
- Fix typo
- Add verboseness if skipping module
- Capture `self.res`
- Minor updates to logging

### CODE REVIEW

**Review by npmitche — APPROVED** [review]

File: chipsec/modules/tools/cpu/sinkhole.py [hunk 1]
```diff
@@ -42,19 +41,24 @@ def __init__(self):
...
 
     def check_LAPIC_SMRR_overlap( self ):
         if not self.cs.is_register_defined( 'IA32_APIC_BASE' ) or \
            not self.cs.is_register_defined( 'IA32_SMRR_PHYSBASE' ) or \
            not self.cs.is_register_defined( 'IA32_SMRR_PHYSMASK' ):
-            self.logger.error( "Couldn't find definition of required configuration registers" )
+            self.logger.error("Couldn't find definition of required configuration registers")
             return ModuleResult.ERROR
 
         if self.cs.cpu.check_SMRR_supported():
```
Comment (BrentHoltsclaw) [inline chipsec/modules/tools/cpu/sinkhole.py:58]: Shouldn't this check be in is_supported?

File: chipsec/modules/tools/cpu/sinkhole.py [hunk 2]
```diff
@@ -70,28 +74,28 @@ def check_LAPIC_SMRR_overlap( self ):
...
-        self.logger.log( "    writing 0x{:X} to IA32_APIC_BASE[APICBase]..".format(smrrbase) )
+        self.logger.log("[*] Attempting to overlap Local APIC page with SMRR region")
+        self.logger.log( "    Writing 0x{:X} to IA32_APIC_BASE[APICBase]..".format(smrrbase) )
         self.logger.log_important( "NOTE: The system may hang or process may crash when running this test. In that case, the mitigation to this issue is likely working but we may not be handling the exception generated.")
         try:
             self.cs.write_register_field( 'IA32_APIC_BASE', 'APICBase', smrrbase, preserve_field_position=False, cpu_thread=0 )
             ex = False
-            self.logger.log_bad( "Could modify IA32_APIC_BASE to overlap SMRR" )
-        except chipsec.helper.oshelper.HWAccessViolationError:
+            self.logger.log_bad("Could modify IA32_APIC_BASE to overlap SMRR")
```
Comment (BrentHoltsclaw) [inline chipsec/modules/tools/cpu/sinkhole.py:83]: Suggest moving this down to else case (line 95).  I am seeing on a recent platform that an error is not thrown, however the address cannot be changed.

File: chipsec/modules/tools/cpu/sinkhole.py [hunk 3]
```diff
@@ -70,28 +74,28 @@ def check_LAPIC_SMRR_overlap( self ):
...
 
         apic_base_msr_new = self.cs.read_register( 'IA32_APIC_BASE', 0 )
-        self.logger.log( "[*] new IA32_APIC_BASE: 0x{:016X}".format(apic_base_msr_new) )
+        self.logger.log( "[*] New IA32_APIC_BASE: 0x{:016X}".format(apic_base_msr_new) )
 
-        if apic_base_msr_new == apic_base_msr and ex:
+        if (apic_base_msr_new == apic_base_msr) and ex:
             res = ModuleResult.PASSED
-            self.logger.log_passed_check( "CPU does not seem to have SMM memory sinkhole vulnerability" )
+            self.logger.log_passed("CPU does not seem to have SMM memory sinkhole vulnerability")
```
Comment (BrentHoltsclaw) [inline chipsec/modules/tools/cpu/sinkhole.py:93]: Suggest adding another case where `apic_base_msr_new == apic_base_msr` and an error was not thrown.

File: chipsec/modules/tools/cpu/sinkhole.py [hunk 4]
```diff
@@ -42,19 +41,24 @@ def __init__(self):
...
-        return (self.cs.helper.is_windows() or self.cs.helper.is_linux())
+        if not (self.cs.helper.is_windows() or self.cs.helper.is_linux()):
+            self.logger('Not a supported OS: Windows or Linux')
+            self.res = ModuleResult.NOTAPPLICABLE
+            return False
+        else:
+            return True
 
     def check_LAPIC_SMRR_overlap( self ):
         if not self.cs.is_register_defined( 'IA32_APIC_BASE' ) or \
```
Comment (BrentHoltsclaw) [inline chipsec/modules/tools/cpu/sinkhole.py:52]: Shouldn't this be part of is_supported?

File: chipsec/modules/tools/cpu/sinkhole.py (hunk 2 above)
Comment (frinzell) [reply chipsec/modules/tools/cpu/sinkhole.py:83]: An error most likely IS thrown but caught further upstream due to #1177

**Review by Sae86 — APPROVED** [review]

**Pull request merged**